### ISSUE TITLE
running with Python2 gives "module 'os' has no attribute 'popen4'

### ISSUE THREAD

**Issue by AstroriaLunaris** [ssb22/lexconvert#2]
tried running the .py file with the arguments `--phones dectalk test` and it returned an error ending in `AttributeError: module 'os' has no attribute 'popen4'` maybe Im missing something here? there are no dependencies listed in the readme file so I have no idea if this is just because I don't have everything I need to actually run this

**Comment by ssb22** [comment]
Hi, the error you got should be impossible with the latest version of lexconvert: it explicitly catches that error and tries an alternative way to get that function.  Please can you check which version of lexconvert you have, it should be version 0.35.  Would you also be able to tell me the exact version of Python you have, and also which operating system you are on?  Thanks.

**Comment by AstroriaLunaris** [comment]
Hi, 
I just re-downloaded it (version 0.35) and I now get the same error regardless of which version of Python I run it in: `TypeError: popen4() takes no keyword arguments` this is with versions 3.9.6 and 2.7.18

**Comment by ssb22** [comment]
Thanks for showing me the actual command output.  The `py` command on Windows is a Python launcher that can launch multiple different versions of Python.  Running `py --version` just tells you which version of Python shipped the `py` command itself; it does not tell you which version it will use when you run a script.  If you have installed multiple versions of Python on your system, then there are several different ways of setting which one will be run by the launcher.

In Python 3 the `popen4` function has been completely removed, which means we'll get the original `AttributeError`, which in the latest version of lexconvert will be caught and redirected to alternate code to handle it in Python 3's way of doing things.  That's not what's happening in your output, so I don't think your output really came from Python 3.  I think you have a `py` command from Python 3.9, but your system has set it to use Python 2 by default.

More worrying is why you're getting a `TypeError` from Python 2.  That `popen4` function has existed in every version of the Python Standard Library from 2.0 through 2.7, and in all of them it took keyword arguments, on both Unix and Windows.  There is no way you'd get `popen4() takes no keyword arguments` from a standard install of any Python 2.x version.  Either something has installed a site startup script that's seriously messing up your `os` module, or else something has installed a custom version of `os.py` itself.

At this point I'm tempted to recommend just deleting all versions of Python from your system and reinstalling it from `python.org`, because something's messed it up and I wouldn't know how to start figuring that out unless I had the computer in front of me (and even then it would be awkward).  I could probably do a lexconvert patch to work around this particular problem, but if that computer's Python is messed up then I expect we'd just hit another problem immediately after solving this one, so really I think it would be better if you could reinstall Python.

**Comment by ssb22** [comment]
Actually there might be a way you'd get that error from some 2.x installs on Windows that use a native version of `popen4` in a DLL: it seems not all Windows versions set the arguments to be keyword arguments as documented.  (This is a Python bug, which there's no point reporting now because the whole thing has been removed in Python 3 anyway.)

Unfortunately, it also seems that the affected versions would not have been able to take the buffer size we're trying to give it.

If I change that function around to prefer the `subprocess` way of doing things, then at least it should work on Python 2.4 and above.  Perhaps I should do that anyway because I can't remember when was the last time I tested this code on Python 2.3 (unlike Gradint itself which I have to keep compatible with Python 2.1 for the S60 port).
